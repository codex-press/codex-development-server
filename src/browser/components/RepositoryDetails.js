import React from 'react';

import { openFileSystem } from '../utility';


export default function RepositoryDetails(props) { 

  let open = path => openFileSystem(props.path + '/' + path);

  return (
    <div className="Repository">

      <h4 className="name">{ props.name }</h4>

      <div className="path" onClick={ () => openFileSystem(props.path) }>
        { props.path }
      </div>

      <div className="collapse" onClick={ props.collapse }>
        see less
      </div>

      <button onClick={ props.remove }>Remove</button>
      {' '}
      <button onClick={ props.edit }>Edit</button>

      <ul>
        { props.files.map(f => 
          <li key={ f.filename }>
            { f.filename + ' ' }
            <span className="open" onClick={ () => open(f.filename) }>(open)</span>
          </li>)
        }
      </ul>

    </div>
  );
}


