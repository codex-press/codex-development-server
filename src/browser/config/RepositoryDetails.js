import React from 'react';

import { openFileSystem } from '../utility';


export default function RepositoryDetails(props) { 

  let open = path => openFileSystem(props.path + '/' + path);

  return (
    <div className="Repository">

      <h4 className="name">{ props.name }</h4>

      <div className="path" onClick={ () => openFileSystem(props.path) }>
        { props.path }
      </div>

      <div className="collapse" onClick={ props.collapse }>
        see less
      </div>

      <button onClick={ props.remove }>Remove</button>
      {' '}
      <button onClick={ props.edit }>Edit</button>

      <table>
        <thead>
          <tr>
            <th>Filename</th>
            <th>Path</th>
          </tr>
        </thead>
        <tbody>
          { props.files.map(f => 
            <tr key={ f.filename }>
              <td>
                <span className="open" onClick={ () => open(f.filename) }>
                  { f.filename + ' ' }
                </span>
              </td>
              <td>
                <a target="_blank" href={ f.path }>
                  { f.path }
                </a>
              </td>
            </tr>)
          }
        </tbody>
      </table>

    </div>
  );
}


