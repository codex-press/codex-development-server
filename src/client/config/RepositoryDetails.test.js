import React from 'react'
import { shallow } from 'enzyme'

import RepositoryDetails from './RepositoryDetails'

const repository = {
  name: "parent",
  path: "/Users/omar/code/codex_press/parent",
  assets: [
    "/parent/index.css",
    "/parent/index.js",
  ],
  files: [
    {
      filename: "index.js",
      path: "/parent/index.js"
    },
    {
      filename: "index.less",
      path: "/parent/index.css",
      deps: [
        "css/editor_nag.less",
        "css/scroll_alert.less",
        "css/article.less",
        "css/knot_loading_graphic.less",
      ]
    },
    {
      filename: "package.json"
    },
    {
      filename: "test/dom/attributes.js"
    },
    {
      filename: "test/dom/classes.js"
    },
  ]
}


test('renders RepositoryDetails', () => {
  const c = shallow(<RepositoryDetails { ...repository } />)
  expect(c).toMatchSnapshot()
});


