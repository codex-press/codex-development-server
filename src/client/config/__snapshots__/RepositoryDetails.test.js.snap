// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders RepositoryDetails 1`] = `
<div
  className="Repository"
>
  <h4
    className="name"
  >
    parent
  </h4>
  <div
    className="path"
    onClick={[Function]}
  >
    /Users/omar/code/codex_press/parent
  </div>
  <div
    className="collapse"
  >
    see less
  </div>
  <button>
    Remove
  </button>
   
  <button>
    Edit
  </button>
  <table>
    <thead>
      <tr>
        <th>
          Filename
        </th>
        <th>
          Path
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <span
            className="open"
            onClick={[Function]}
          >
            index.js 
          </span>
        </td>
        <td>
          <a
            href="/parent/index.js"
            target="_blank"
          >
            /parent/index.js
          </a>
        </td>
      </tr>
      <tr>
        <td>
          <span
            className="open"
            onClick={[Function]}
          >
            index.less 
          </span>
        </td>
        <td>
          <a
            href="/parent/index.css"
            target="_blank"
          >
            /parent/index.css
          </a>
        </td>
      </tr>
      <tr>
        <td>
          <span
            className="open"
            onClick={[Function]}
          >
            package.json 
          </span>
        </td>
        <td>
          <a
            target="_blank"
          />
        </td>
      </tr>
      <tr>
        <td>
          <span
            className="open"
            onClick={[Function]}
          >
            test/dom/attributes.js 
          </span>
        </td>
        <td>
          <a
            target="_blank"
          />
        </td>
      </tr>
      <tr>
        <td>
          <span
            className="open"
            onClick={[Function]}
          >
            test/dom/classes.js 
          </span>
        </td>
        <td>
          <a
            target="_blank"
          />
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
